core/src/main/scala/org/apache/spark/sql/delta/commands/DeleteCommand.scala

```diff
@@ -93,9 +116,12 @@ case class DeleteCommand(
       // Re-cache all cached plans(including this relation itself, if it's cached) that refer to
       // this data source relation.
       sparkSession.sharedState.cacheManager.recacheByPlan(sparkSession, target)
+      val knownDeletedRows = deletionResultFromMetrics match {
+        case RowInFilesDeletion(deletedRows) => deletedRows
+        case FilePruningAtPartitionBoundary => -1
+      }
```

Comment:
I think using these traits and case classes is more complex than necessary (we should be able to do this in a few lines of code).

I don't see why we can't just check if numDeletedRows = 0 and numRemovedFiles > 0 and then update our result to be -1 if so. A comment here explaining that this is because it's a delete over partitions and we can't know the number of deleted rows would suffice.

Reply:
Hey I saw your slack comment. I personally feel something like this is much easier to read, especially with the nested if statements you have above. I agree with the principles you have in mind in regards to self-documenting code, but I think this specific scenario is so simple logic-wise that it's unnecessary and actually more complex to the reader.

```scala
var deletedRows = metrics("numDeletedRows").value
if (metrics("numRemovedFiles").value > 0 && deletedRows == 0) {
    // this is a delete over a partition boundary which is a metadata only operation. therefore we can't know how many rows are deleted
    deletedRows = -1
}
```

Reply:
If you feel really strongly about this, we can look into making your current implementation more readable (it personally took me a few seconds to deduce what was going on)

Reply:
You are more familiar with the codebase and the context than me for sure, so if you think the scenario is simple enough so that separating interpreting metrics from results is not required, I will just remove it.  

The reason why it was surprising to me is that the -1 is sort of a magic value, and I would have expected some sort of metadata to be available to tell how many files rows are deleted in each file